src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -62,7 +73,30 @@ class EntityClient (requestContext: RequestContext) extends Actor {
         requestContext.failWith(
           new RequestProcessingException(StatusCodes.InternalServerError, error.getMessage))
     }
+  }
 
+  def createEntities(wsNamespace: String, wsName: String, entities: Seq[Entity]) = {
+    val results = entities.map(entity => createEntityOrReportError(wsNamespace, wsName, entity))
+    // have to manually construct an HttpResponse here because we're aggregating the results of multiple calls to Rawls
+    val response = HttpResponse(OK, HttpClient.createJsonHttpEntity(results.toJson.compactPrint))
+    requestContext.complete(response)
+  }
+
+  def createEntityOrReportError(workspaceNamespace: String, workspaceName: String, entity: Entity): EntityCreateResult = {
+    val url = s"${FireCloudConfig.Workspace.entityPathFromWorkspace(workspaceNamespace, workspaceName)}"
+    val externalRequest = Post(url, HttpClient.createJsonHttpEntity(entity.toJson.compactPrint))
+    val pipeline = addHeader(Cookie(requestContext.request.cookies)) ~> sendReceive
+    // TODO add a timeout?
+    // TODO is there a way to return the EntityCreateResult from within an onComplete, instead of doing this?
+    Try(Await.result(pipeline(externalRequest), Duration.Inf)) match {
+      case Success(response) => response.status match {
+        case StatusCodes.Created => EntityCreateResult(entity.entityType, entity.entityName, true, "Entity created successfully")
+        case _ => EntityCreateResult(entity.entityType, entity.entityName, false, s"Bad response from workspace service: ${response.message}")
+      }
+      // TODO how should we handle the case where an exception is thrown?
```

Comment:
There is an example in this class for handling exceptions. Update the logging message if you would like to see more detail
```
      case Failure(error) =>
        // Failure accessing service
        log.error(error, "Service API call failed")
        requestContext.failWith(
          new RequestProcessingException(StatusCodes.InternalServerError, error.getMessage))
```